src/solvers/strings/string_constraint_generator_constants.cpp

```diff
@@ -86,17 +87,19 @@ std::pair<exprt, string_constraintst> add_axioms_for_cprover_string(
   const array_string_exprt &res,
   const exprt &arg,
   const exprt &guard,
-  array_poolt &array_pool)
+  array_poolt &array_pool,
+  const namespacet &ns)
 {
   if(const auto if_expr = expr_try_dynamic_cast<if_exprt>(arg))
   {
     const and_exprt guard_true(guard, if_expr->cond());
     const and_exprt guard_false(guard, not_exprt(if_expr->cond()));
     return combine_results(
       add_axioms_for_cprover_string(
-        fresh_symbol, res, if_expr->true_case(), guard_true, array_pool),
+        fresh_symbol, res, if_expr->true_case(), guard_true, array_pool, ns),
       add_axioms_for_cprover_string(
-        fresh_symbol, res, if_expr->false_case(), guard_false, array_pool));
+        fresh_symbol, res, if_expr->false_case(), guard_false, array_pool, ns),
```

Comment:
It looks like `add_axioms_for_cprover_string` is only called with `guard=true_exprt{}` so you could remove this argument and get rid of this `if`, that will avoid having to introduce the `ns` argument here and in the calling function (and maybe others).

Reply:
That might reduce the arguments by one, but pretty sure everything else is still required? The guard may always be true, but the condition on the if is variable and having to do the branching / merge is still required as well right?

Reply:
oh yes I was mistaken, the recursive call changes the guard